### PR TITLE
added tests for c/mortgage library

### PR DESCRIPTION
**Pull request by pchittum**
Test suite for `c/mortgage` library component.

### CODE REVIEW

**Comment by pchittum** [comment]
**Note:** Instead of a raw number comparison in `calculateMonthlyPayment function returns monthly payment with valid inputs` test, I'm calling `toFixed(2)` on both a test number and the result to ensure a deterministic test that aligns to user experience. Otherwise I was getting a test pass running locally, but the github action was failing with the following: 

```
    Expected: 277.298798926959
    Received: 277.29879892695965
```

File: force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            // invoke with 4 invalid input sets and store each return value in an array
+            const results = invalidInputs.inputs.map((item) =>
+                calculateMonthlyPayment(item.principal, item.years, item.rate)
+            );
+
+            expect(results).toMatchObject(invalidInputs.expectedResults);
+        });
+    });
+
+    it('getTermOptions returns valid term options', () => {
```
Comment (pozil) [inline force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js:34]: @muenzpraeger I'd like your advice on this one.

I don't think that testing this makes sense because we're duplicating constant data in our test data. There's no particular logic under test here.
IMHO the `mortage` implementation should be changed to just export a constant instead of a function:

```js
const TERM_OPTIONS = [ ... ];
...
export { TERM_OPTIONS, calculateMonthlyPayment };
```

File: force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+            const monthlyPayment = calculateMonthlyPayment(
+                principal,
+                years,
+                rate
+            );
+            // convert long decimal to two decimal places and compare
+            expect(monthlyPayment.toFixed(2)).toBe(expectedResult);
+        });
+        it('returns zero with invalid inputs', () => {
```
Comment (pozil) [inline force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js:21]: Add line break

File: force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js (hunk 1 above)
Comment (pchittum) [reply force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js:34]: I agree, actually. This test is pointless. And the actual use of it in `c/miscSharedJavaScript` should also be implemented as a getter (instead of a property pass through). 

I'll add this to the work expected in issue #323. 

I can leave this for now, and refactor later when I come back.

Comment (pchittum) [reply force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js:34]: My initial thinking was to agree with you. 

But is this intending to mock what would often be a dynamic set of picklist values? If so, then I'd argue with keeping the test case.

Comment (muenzpraeger) [reply force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js:35]: At least we should test the default values. Agreeing with @pchittum.

File: force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                years,
+                rate
+            );
+            // convert long decimal to two decimal places and compare
+            expect(monthlyPayment.toFixed(2)).toBe(expectedResult);
+        });
+
+        it('returns zero with invalid inputs', () => {
+            // each value tests each param as falsy, third param also tested for explicit zero value
+            const invalidInputs = require('./data/invalidInputsTestData.json');
```
Comment (muenzpraeger) [inline force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js:24]: We should move this outside the test body.

File: force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import { getTermOptions, calculateMonthlyPayment } from 'c/mortgage';
+
+describe('c-mortgage', () => {
+    describe('calculateMonthlyPayment function', () => {
+        it('returns monthly payment with valid inputs', () => {
+            const principal = 50000;
```
Comment (muenzpraeger) [inline force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js:6]: If it's a const for representing fixed test value it should be `const PRINCIPAL = ...` (same for subsequent variables).

File: force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js (hunk 3 above)
Comment (pchittum) [reply force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js:24]: This entire data set is only applicable to this test. Moving it outside will make the test less clear.

File: force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js (hunk 4 above)
Comment (pchittum) [reply force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js]: fixed

File: force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js (hunk 3 above)
Comment (muenzpraeger) [reply force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js:24]: This it. Then it should be moved to `const INVALID_INPUTS` at the top of the test, to make it consistent. Also, does the test cover cases where non-numeric data gets passed to the component (i. e. a String)?

Comment (pchittum) [reply force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js:24]: The function doesn't implement any specific logic around non-numeric data. This test is to handle the specific line of logic that returns zero when falsy values (or zero in the case of rate) are passed in. 

Since this is a service function, error handling would be handled up the stack (presumably) as it was not implemented here.

File: force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js (hunk 2 above)
Comment (pchittum) [reply force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js:22]: this is done.

Comment (pchittum) [reply force-app/main/default/lwc/mortgage/__tests__/mortgage.test.js:22]: Done.

**Comment by pchittum** [comment]
Never mind on my request to re-review @muenzpraeger. I should have checked which change requested was still outstanding.

**Comment by pchittum** [comment]
@pozil I think I addressed your changes. Can you let me know otherwise, or approve?
